### ISSUE TITLE
process.versions.electron can be overridden

### ISSUE THREAD

**Issue by MarshallOfSound** [electron/electron#8083]
* Electron version: All
* Operating system: All

### Expected behavior

`process.versions.electron` should be non-configurable

### Actual behavior

```js
process.versions.electron = '99.99.99'
```

Code such as the above will override the Electron version which can stop third party modules correctly recognising the Electron version.

### How to reproduce

Run the code above^^

The reason this is an issue is that other versions such as `process.versions.node` can not be configured / changed.

**Comment by MarshallOfSound** [comment]
I looked into fixing this but it doesn't look like the native mate `Dictionary` class has methods to set non-configurable properties.  I would do it in JS but I think it would be better to possibly add methods to native mate to do this.

/cc @zcbenz

**Comment by tonyganch** [comment]
Do I get it right that the issue is fixed already in master?

**Comment by MarshallOfSound** [comment]
> Do I get it right that the issue is fixed already in master?

It will be fixed in the 2.0 update as it is a breaking change

**Comment by tonyganch** [comment]
Should this issue be closed as fixed, even though it's not released yet?

**Comment by kevinsawicki** [comment]
> Should this issue be closed as fixed, even though it's not released yet?

No, it will need to be further updated in a separate pull request.

**Comment by tonyganch** [comment]
@MarshallOfSound, can this issue be closed finally?

**Comment by MarshallOfSound** [comment]
@tonyganch Idk, did a PR fix it 🤔 Can't see anything reffing this issue
